.github/workflows/build-and-test-all.yml

```diff
@@ -482,19 +482,12 @@ jobs:
       - test-recursor-api
       - test-recursor-regression
       - test-recursor-bulk
+    if: success() || failure()
     runs-on: ubuntu-20.04
     steps:
-      - uses: actions/checkout@v3.1.0
-        with:
-          fetch-depth: 5
-          submodules: recursive
-      - name: Install yq
-        run: sudo wget https://github.com/mikefarah/yq/releases/download/v4.9.6/yq_linux_amd64 -O /usr/bin/yq && sudo chmod +x /usr/bin/yq
-      - name: Get full list of jobs for this workflow
-        run: yq e '.jobs | keys' .github/workflows/build-and-test-all.yml | grep -v '^- collect' | sort | tee /tmp/workflow-jobs-list.yml
-      - name: Get list of jobs the collect job depends on
-        run: yq e '.jobs.collect.needs | ... comments=""' .github/workflows/build-and-test-all.yml | sort | tee /tmp/workflow-collect-dependencies.yml
-      - name: Diff them
-        run: diff -u /tmp/workflow-jobs-list.yml /tmp/workflow-collect-dependencies.yml
+      - name: Install jq
+        run: "sudo DEBIAN_FRONTEND=noninteractive apt-get install -qq -y jq"
+      - name: Fail job if any of the previous jobs failed
+        run: "for i in `echo '${{ toJSON(needs) }}' | jq '.[].result' | tr -d '\"'`; do if [[ $i == 'failure' ]]; then echo '${{ toJSON(needs) }}'; exit 1; fi; done;"
```

Comment:
If we ever add a job, and forget to add it to `needs`, you're likely to hit that job as "not yet succeeded" here I think. So I think the diff (or something like it) might still be necessary?

Reply:
Sure, we could add a third step for checking other jobs not included in the needs section. However, bear in mind that if that job has not been included in the needs section, it could (potentially) still be running when the check for other jobs is executed.

Reply:
Right, I see my first comment was incorrect. Your check looks at `needs` so it would never run into the value. Then indeed we need something that checks whether `needs` is still complete.

Reply:
Added check to fail job also if the list of prerequired jobs is not complete (new jobs added and not reflected there).